packages/backend/src/functions/socketServer/SocketServer.ts

```diff
@@ -0,0 +1,87 @@
+import socketio from "socket.io";
+import http from "http";
+import { Terminal } from "../terminal";
+import { FileSystem } from "../filesystem";
+import Project from "../Project";
+import fsTree from "../fsTree";
+import GitHub from "../oauth/GitHub";
+import { User as GitHubUser } from "malte-common/dist/oauth/GitHub";
+import { isUser } from "malte-common/dist/oauth/isUser";
+
+export default class SocketServer {
+  protected static instance: SocketServer;
+
+  protected project: Project;
+  public server: SocketIO.Server;
+
+  protected userMap = new Map<string, GitHubUser>();
+
+  protected setUpEvents(): void {
+    this.server.on("connection", this.onConnection.bind(this));
+  }
+
+  private onConnection(socket: SocketIO.Socket): void {
+    // everyone must be able to request to join, otherwise noone can join
```

Comment:
```suggestion
    // everyone must be able to request to join, otherwise no one can join
```

See https://www.grammarly.com/blog/no-one-noone/